### DISCUSSION TITLE
CSV reader - different number of columns

### DISCUSSION THREAD

**Discussion by dberardo-com** [duckdb/duckdb#8846] [Q&A]
is it possible csv files that have variable number of columns in them? like

a;b;c
1;2;3
1;2
0;3;2

and simply fill with null the missing columns ?

same question would apply for json files.

**Comment by dberardo-com** [comment]
this seems to be already available in case of file union: https://duckdb.org/docs/data/csv/tips#use-union_by_name-when-loading-files-with-different-schemas how about single file read ?

**Comment by pdet** [comment]
With CSV files you can use the `null_padding=true` option to null pad missing columns.
